src/main/java/org/cryptomator/cryptofs/event/FilesystemEvent.java

```diff
@@ -0,0 +1,27 @@
+package org.cryptomator.cryptofs.event;
+
+import java.util.function.Consumer;
+
+/**
+ * Common interface for all filesystem events.
+ * <p>
+ * Events are emitted via the notification method set in the properties during filesystem creation, see {@link org.cryptomator.cryptofs.CryptoFileSystemProperties.Builder#withFilesystemEventConsumer(Consumer)}.
+ * <p>
+ * To get a specific event type, use the enhanced switch pattern or typecasting in if-instance of, e.g.
+ * {@code
+ * FilesystemEvent fse;
+ * switch (fse) {
+ *   case DecryptionFailedEvent dfe -> //do stuff
+ *   case ConflictResolvedEvent cre -> //do other stuff
```

Comment:
don't encourage people to use weird abbreviations for variables that don't even share the same scope

```suggestion
 *   case DecryptionFailedEvent e -> //do stuff
 *   case ConflictResolvedEvent e -> //do other stuff
```